core/admin/admin.php

```diff
@@ -721,7 +722,14 @@ public function admin_action_new_post() {
 			wp_die( $document ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		}
 
-		wp_redirect( $document->get_edit_url() );
+		$active_document = Utils::get_super_global_value( $_GET, 'active-document' ) ?? null;
+		$edit_url = $document->get_edit_url();
+
+		if ( $active_document ) {
+			$edit_url = add_query_arg( 'active-document', $active_document, $edit_url );
+		}
```

Comment:
@mykytamurzin Can we rather do this in the home screen `module.php` file?
Filter: `elementor/document/urls/edit`